integration-tests/src/test/java/org/apache/druid/tests/coordinator/duty/ITAutoCompactionTest.java

```diff
@@ -0,0 +1,351 @@
+      intervalsBeforeCompaction.sort(null);
+      // 4 segments across 2 days (4 total)...
+      verifySegmentsCount(4);
+      verifyQuery(INDEX_QUERIES_RESOURCE);
+
+      submitCompactionConfig(MAX_ROWS_PER_SEGMENT_COMPACTED, SKIP_OFFSET_FROM_LATEST);
+      deleteCompactionConfig();
+      forceTriggerAutoCompaction();
+
+      // ...should remains unchanged (4 total)
+      verifySegmentsCount(4);
+      verifyQuery(INDEX_QUERIES_RESOURCE);
+      verifySegmentsCompacted(0, null);
+
+      checkCompactionIntervals(intervalsBeforeCompaction);
+    }
+  }
+
+  @Test
+  public void testAutoCompactionDutyCanUpdateTaskSlots() throws Exception
+  {
+    loadData(INDEX_TASK);
+    try (final Closeable ignored = unloader(fullDatasourceName)) {
+      final List<String> intervalsBeforeCompaction = coordinator.getSegmentIntervals(fullDatasourceName);
+      intervalsBeforeCompaction.sort(null);
+      // 4 segments across 2 days (4 total)...
+      verifySegmentsCount(4);
+      verifyQuery(INDEX_QUERIES_RESOURCE);
+
+      // Skips first day. Should only compact one out of two days.
+      submitCompactionConfig(MAX_ROWS_PER_SEGMENT_COMPACTED, SKIP_OFFSET_FROM_LATEST);
+
+      // Set compactionTaskSlotRatio to 0 to prevent any compaction
+      updateCompactionTaskSlot(0, 100);
+      forceTriggerAutoCompaction();
+      // ...should remains unchanged (4 total)
+      verifySegmentsCount(4);
+      verifyQuery(INDEX_QUERIES_RESOURCE);
+      verifySegmentsCompacted(0, null);
+      checkCompactionIntervals(intervalsBeforeCompaction);
+
+      // Set maxCompactionTaskSlots to 0 to prevent any compaction
+      updateCompactionTaskSlot(0.1, 0);
+      forceTriggerAutoCompaction();
+      // ...should remains unchanged (4 total)
+      verifySegmentsCount(4);
+      verifyQuery(INDEX_QUERIES_RESOURCE);
+      verifySegmentsCompacted(0, null);
+      checkCompactionIntervals(intervalsBeforeCompaction);
+
+      // Update compaction slots to be 1
+      updateCompactionTaskSlot(1, 1);
+      forceTriggerAutoCompaction();
+      // One day compacted (1 new segment) and one day remains uncompacted. (5 total)
+      verifySegmentsCount(5);
+      verifyQuery(INDEX_QUERIES_RESOURCE);
+      verifySegmentsCompacted(1, MAX_ROWS_PER_SEGMENT_COMPACTED);
+      checkCompactionIntervals(intervalsBeforeCompaction);
+      Assert.assertEquals(compactionResource.getCompactionProgress(fullDatasourceName).get("remainingSegmentSize"), "14312");
+      // Run compaction again to compact the remaining day
+      forceTriggerAutoCompaction();
+      // Remaining day compacted (1 new segment). Now both days compacted (6 total)
+      verifySegmentsCount(6);
+      verifyQuery(INDEX_QUERIES_RESOURCE);
+      verifySegmentsCompacted(2, MAX_ROWS_PER_SEGMENT_COMPACTED);
+      checkCompactionIntervals(intervalsBeforeCompaction);
+//      Assert.assertEquals(compactionResource.getCompactionProgress(fullDatasourceName).get("remainingSegmentSize"), "0");
```

Comment:
@jihoonson Not sure if this is expected but when I call the API for get compaction progress here I got status[400 Bad Request] content[{"error":"unknown dataSource"}] instead of 0. I expected 0 since all segments of the datasource have been compacted.